pkg/provider/branch/trunk/trunk.go

```diff
@@ -398,12 +400,14 @@ func (t *trunkENI) CreateAndAssociateBranchENIs(pod *v1.Pod, securityGroups []st
 		newENIs = append(newENIs, newENI)
 
 		// Associate Branch to trunk
-		_, err = t.ec2ApiHelper.AssociateBranchToTrunk(&t.trunkENIId, nwInterface.NetworkInterfaceId, vlanID)
+		var associationOutput *awsEC2.AssociateTrunkInterfaceOutput
```

Comment:
Can we remove this statement and directly assigning the returned value to the variable in the next line?

Reply:
The declaration here needs to be separate. We do not want to re-declare `err` here since we use it in the next check to see if creation/association of branch ENIs failed in which case we move all branch ENIs to the delete queue.

Reply:
I don't think that is how error works here. If create fails, error leads to break the loop already. If error reach here, it is nil. Here we check error returned from associate call. If it is not nil, we break from the loop. Actually we should avoid reusing err if possible.
err = createNetworkInterface()
if err != nil break
else
...
err = associateNetworkInterface()
if err != nil break
else
...

Reply:
Synced offline on this; this cannot be moved to short assignment as `err` will be re-declared and we miss catching issues with association. Resolving this.